### PR TITLE
MGMT-5449 Provide Minimum HW requirements for SNO

### PR DESCRIPTION
**Pull request by tsorya**
From my cluster :

Minimum hardware requirements
One host is required with at least 8 CPU cores, 32.00 GiB of RAM, and 10 GB of filesystem storage.

### CODE REVIEW

File: internal/hardware/validator.go [hunk 1]
```diff
@@ -219,10 +227,20 @@ func (v *validator) getPreflightOCPRequirements(cluster *common.Cluster) (*model
 	}, nil
 }
 
-func (v *validator) getOCPRequirementsForVersion(openShiftVersion string) *models.VersionedHostRequirements {
-	requirements, err := v.VersionedRequirements.GetVersionedHostRequirements(openShiftVersion)
+func (v *validator) updateSingleNodeHwRequirements(requirements *models.VersionedHostRequirements) {
+	requirements.MasterRequirements.CPUCores = v.ValidatorCfg.MinCPUCoresSno
+	requirements.MasterRequirements.RAMMib = conversions.GibToMib(v.ValidatorCfg.MinRamGibSno)
```
Comment (eranco74) [inline internal/hardware/validator.go:232]: Shre this will do the trick?
Currently, it seems that the minimum requirement is the worker/default 8GiB

**Comment by romfreiman** [comment]
@ori-amizur @filanov @ronniel1 @gamli75 wdyt?

File: internal/hardware/validator.go (hunk 1 above)
Comment (tsorya) [reply internal/hardware/validator.go:232]: yeah it will, tested it

File: swagger.yaml [hunk 2]
```diff
@@ -2985,6 +2985,13 @@ paths:
         - userAuth: [admin, read-only-admin, user]
       description: Get minimum host requirements.
       operationId: GetHostRequirements
+      parameters:
```
Comment (filanov) [inline swagger.yaml:2988]: maybe instead of having another parameter you could add it `host-requirements`? you have master/worker/sno

Comment (eranco74) [reply swagger.yaml:2988]: Why `in: header`?

Comment (tsorya) [reply swagger.yaml:2988]: thought about it , but i was not sure what it break. So prefered smaller change but will open ticket for this one.

Comment (filanov) [reply swagger.yaml:2988]: You are adding parameters so it should not break anything

Comment (filanov) [reply swagger.yaml:2988]: I think that code will be simple as well.

Comment (tsorya) [reply swagger.yaml:2988]: to say the truth i don't know who uses this api at all

Comment (filanov) [reply swagger.yaml:2988]: @avishayt what do you think?

Comment (avishayt) [reply swagger.yaml:2988]: @tsorya @filanov I'm fairly sure this API is no longer used.  The new APIs are:
/clusters/{cluster_id}/host-requirements
/clusters/{cluster_id}/preflight-requirements

In both cases there should be no API change, just check if the cluster is SNO then return the appropriate requirements.

**Pull request merged**